### PR TITLE
Add rule for preferring to syntax for creating pairs

### PR DESCRIPTION
**Pull request by jkaan**
This implements the rule talked about in #912

### CODE REVIEW

File: detekt-rules/src/test/resources/cases/PreferToOverPairSyntaxPositive.kt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package cases
+
+@Suppress("unused", "UNUSED_VARIABLE")
+fun preferToOverPairSyntaxPositive() {
+	val pair1 = Pair(1, 2)
+
+	val pair2: Pair<Int, Int> = Pair(1, 2)
```
Comment (schalkms) [inline detekt-rules/src/test/resources/cases/PreferToOverPairSyntaxPositive.kt]: Please also test this also with a nested pair
e.g. a pair element which contains another pair

File: detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/optional/PreferToOverPairSyntax.kt [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+	override fun visitSimpleNameExpression(expression: KtSimpleNameExpression) {
+		val callReference = expression.getQualifiedElementOrCallableRef()
+		if (expression.getReferencedName() == PAIR_CONSTRUCTOR_REFERENCE_NAME &&
+				callReference is KtCallExpression) {
+			val (firstArg, secondArg) = callReference.valueArguments.map { it.text }
+
+			report(CodeSmell(issue, Entity.from(expression),
+					message = "Pair is created by using the pair constructor. " +
+							"This can replaced by $firstArg to $secondArg"))
```
Comment (Mauin) [inline detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/optional/PreferToOverPairSyntax.kt]: can we add backticks into this message? `` `$firstArg to $secondArg` ``

File: detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/optional/PreferToOverPairSyntax.kt [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+		}
+
+		super.visitSimpleNameExpression(expression)
+	}
+
+	companion object {
+		const val PAIR_CONSTRUCTOR_REFERENCE_NAME = "Pair"
```
Comment (Mauin) [inline detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/optional/PreferToOverPairSyntax.kt]: Only issue I can see is codebases that have a custom `Pair` class might accidentally trigger this rule even though their `Pair` doesn't support the `to` notation. But in that case users should just turn off this rule 👍

**Comment by vanniktech** [comment]
Another rule that would benefit from autocorrect.

**Pull request merged**